src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_NUMBER_OF_FIGURES = 5;
+
+    private static boolean isDefault = true;
+    private int index;
+    private Random random = new Random();
+    private Figure defaultFigure;
+    private Figure randomFigure;
+
+    public static boolean isDefault() {
+        return isDefault;
+    }
+
+    public Figure getRandomFigure() {
+        isDefault = false;
+        index = random.nextInt(MAX_NUMBER_OF_FIGURES);
+
+        switch (index) {
+            case 0 :
+                randomFigure = new Square();
+                break;
+            case 1 :
+                randomFigure = new Rectangle();
+                break;
+            case 2 :
+                randomFigure = new RightTriangle();
+                break;
+            case 3 :
+                randomFigure = new IsoscelesTrapezoid();
+                break;
+            case 4 :
+            default:
```

Comment:
It will be better to create separate case 5 for Circle, and in default block throw an exception or call getDefaultFigure() method to make code consistent